README.md

```diff
@@ -173,6 +173,38 @@ The Geolite2-City database can be downloaded from MaxMind's site at [https://dev
 Download the GeoIP2 Binary and save it at `geo_data/GeoLite2-City.mmdb`.
 The app will start using that Geolite2 file for geolocation after restart.
 
+#### Testing on a mobile device or in a virtual machine
+
+By default, the application binds to `localhost`. To test on a network device or within a virtual machine, you can bind to `0.0.0.0` instead, using the following instructions:
+
+1. Determine your computer's network IP address. On macOS, you can find this in the "Network" system settings, shown under the "Status: Connected" label. This often takes the format of `192.168.1.x` or `10.0.0.x`.
+2. In `config/application.yml`, replace `localhost` in the `domain_name` setting with the IP address discovered in the previous step. Leave the trailing port `:3000` unchanged.
+3. Start the server using the command `SMTP_HOST=192.168.1.x rails s -b 0.0.0.0`, replacing `SMTP_HOST` value with the IP address from the first step.
+  - Note that because this uses the Rails built-in server and bypasses the Foreman process, ancillary services like Mailcatcher and Webpack development server will not be started.
```

Comment:
How can we avoid needing to run the built-in server and just let `make run` work as-is?

One can edit Procfile locally, but since the file is tracked by git, it's a very non-ideal workflow to make sure those changes aren't accidentally committed.

```diff
diff --git a/Procfile b/Procfile
index bce413c80..73a5dc488 100644
--- a/Procfile
+++ b/Procfile
@@ -1,3 +1,3 @@
-web: bundle exec rackup config.ru --port ${PORT:-3000}
+web: bundle exec rackup config.ru --port ${PORT:-3000} --host 0.0.0.0
 mailcatcher: mailcatcher -f
 webpacker: ./bin/webpack-dev-server
```

If we want to keep the default `localhost`, maybe we could at least make it overridable by environment variable, like with the `PORT` value?

```diff
diff --git a/Procfile b/Procfile
index bce413c80..cc62ec291 100644
--- a/Procfile
+++ b/Procfile
@@ -1,3 +1,3 @@
-web: bundle exec rackup config.ru --port ${PORT:-3000}
+web: bundle exec rackup config.ru --port ${PORT:-3000} --host ${HOST:-localhost}
 mailcatcher: mailcatcher -f
 webpacker: ./bin/webpack-dev-server
```

Then `HOST=0.0.0.0 make run` can work.

Reply:
Yes, big fan of the second option

Reply:
```suggestion
  - Note that because this uses the Rails built-in server and bypasses the Foreman process, the Mailcatcher process will not be started.
```

I believe `rails s` starts the webpack dev server?

I primarily use `rails s` and haven't had any issues with JS not being recompiled.

Reply:
I need to double-check, because I was also confused about this. I _think_ the distinction is that Rails will recompile JavaScript, but the `webpack-dev-server` process handles automatically refreshing the page after compilation completes.